### ISSUE TITLE
CR Request for Decentralized Identifiers (DIDs) v1.0

### ISSUE THREAD

**Issue by iherman** [w3c/transitions#320]
# Document title, URLs, estimated publication date

Decentralized Identifiers (DIDs) v1.0
- Document on TR: https://www.w3.org/TR/2021/CR-did-core-20210318/
- Editor's draft: https://w3c.github.io/did-core/
- Expected publication date: 2021-03-18

# Abstract

See https://www.w3.org/TR/2021/CR-did-core-20210318/#abstract

# Status

See https://www.w3.org/TR/2021/CR-did-core-20210318/#sotd

# Link to group's decision to request transition

https://www.w3.org/2019/did-wg/Meetings/Minutes/2021-03-09-did#resolution1

# Changes

The document has been significantly restructured and added onto since the First Public Working Draft.  The major changes are:
- the addition of the Abstract Data Model
- new JSON and CBOR representations
- abstract functions describing DID resolution and DID URL dereferencing

For a detailed list of changes, see https://www.w3.org/TR/2021/CR-did-core-20210318/#changes

# Requirements satisfied

The working group believes we have enabled the core use cases described in this document: https://www.w3.org/TR/did-use-cases/

The chairs also believe that this document is one of the best of its kind and should be offered as an example to other groups.

# Dependencies met (or not)

N/A

# Wide Review

Requests for review were sent out to the relevant communities in the space, including
- W3C Credentials Community Group (CCG)
- Decentralized Identity Foundation (DIF)
- Hyperledger Indy and Aries

Additionally, required horizontal reviews were provided by:
- PING:  https://github.com/w3c/did-core/issues/291#issuecomment-785439265
- Security: https://github.com/w3c/did-core/issues/291#issuecomment-786722078
- WAI: https://github.com/w3c/did-core/issues/105#issuecomment-655616261
- I18N: https://github.com/w3c/did-core/issues/104#issuecomment-649187234
- TAG: https://github.com/w3c/did-core/issues/119#issuecomment-763945275

All reviews, comments, etc., from all sources, were tracked on Github issues; see below.

# Issues addressed

The following is the status at the time of submission of this transition request.

The GitHub repository for issues is at https://github.com/w3c/did-core/issues  
All issues blocking CR entry, regardless of source, have been addressed.

A complete disposition of pre-CR comments can be found here:
https://lists.w3.org/Archives/Public/public-did-wg/2021Mar/att-0003/01-part

The following categories of issues remain open:
- Issues that are deferred to a future version of the specification: 
  https://github.com/w3c/did-core/issues?q=is%3Aissue+is%3Aopen+label%3Adefer-v2
- Issues that are purely editorial:
  https://github.com/w3c/did-core/issues?q=is%3Aissue+is%3Aopen+label%3Aeditorial
- Other issues that came in after the WG froze the document for CR vote: 
    - https://github.com/w3c/did-core/issues/707 - "CR Feedback: Support of `publicKeyMultibase`" - The DID WG requested feedback on the `publicKeyMultibase` property via a "Feature at Risk" issue marker. This issue captures this feedback.
    - https://github.com/w3c/did-core/issues/708 - "verification method IDs MUST contain a fragment" - The person that filed the issue seems to have made an invalid assumption about the specification; the specification is correct.

# Formal Objections

None.

# Implementation

From the SOTD: "To exit the W3C Candidate Recommendation phase, the W3C DID Working Group will require two things: 1) for normative statements that are machine testable, at least two interoperable implementations per feature, and 2) for normative statements that are not machine testable, at least two demonstrations of implementation per feature. A feature is defined as one or more functionally related normative statements in the specification."

Implementation Report Instructions: https://w3c.github.io/did-test-suite/  
(in progress)

Test Suite Repository: https://github.com/w3c/did-test-suite  
(in progress)

Expected implementers include: Transmute, Digital Bazaar, Secure Key, Hyperledger Indy, Microsoft, TranSendX

Registered DID Methods: https://w3c.github.io/did-spec-registries/#did-methods  
(registration does not necessarily mean implementation, but it does indicate a high level of interest)

# Patent disclosures

https://www.w3.org/2004/01/pp-impl/117488/status

# Additional information: media type registrations

The document includes the proposals for registration of three media types:

- [application/did+json](https://w3c.github.io/did-core/#application-did-json)
- [application/did+ld+json](https://w3c.github.io/did-core/#application-did-ld-json)
- [application/did+cbor](https://w3c.github.io/did-core/#application-did-cbor)

Mails have been sent to IETF as a request for comments, see:

- https://mailarchive.ietf.org/arch/msg/media-types/p7yMsHEMgG7c0Cf7Oix-59qmNlw/
- https://mailarchive.ietf.org/arch/msg/media-types/QDV_wPk_IimRhLTYGO7ZVJAFg3U/
- https://mailarchive.ietf.org/arch/msg/media-types/YDFxkn9zorm-F4RNZR7ghOwmEP0/

We have received multiple comments to date highlighting concerns related to structured Media Type identifiers. Note that "double +" media types like `application/did+ld+json` are controversial; the group has initiated a separate discussion within IETF:

https://mailarchive.ietf.org/arch/browse/media-types/?gbt=1&index=L0YVPU1zUIlECS8_4ScB5H62GNI

which, so far, has led to a productive discussion without a conclusion to date. At this moment, `application/did+ld+json` is marked as a feature at risk in case that registration is blocked by IANA.

One of the reasons the group is a bit late in the requests for comment mails was due to the pending issue on "double +". The group intends to submit a final submission forms in the coming weeks.

**Comment by iherman** [comment]
Cc: @msporny @rhiaro @burnburn @talltree @brentzundel @peacekeeper

**Comment by plehegar** [comment]
[Diff between WD and ED](https://services.w3.org/htmldiff?doc1=https://www.w3.org/TR/2021/CR-did-core-20210318/&doc2=https://w3c.github.io/did-core/)
 
[Normative references](https://labs.w3.org/normative-references/check?url=https://w3c.github.io/did-core/)

Reviews: [i18n](https://github.com/w3c/i18n-request/issues/120), [APA](https://lists.w3.org/Archives/Public/public-apa/2020May/0038.html), [Privacy](https://lists.w3.org/Archives/Public/public-privacy/2020OctDec/0053.html), [Security](https://lists.w3.org/Archives/Public/public-web-security/2020Dec/0003.html), [TAG](https://github.com/w3ctag/design-reviews/issues/556)

[Horizontal comment tracker](https://www.w3.org/PM/horizontal/review.html?shortname=did-core)

**Comment by swickr** [comment]
Transition approved.

**Comment by iherman** [comment]
Publication done: 

- HPN News: https://www.w3.org/blog/news/archives/8966
- AC announcement: https://lists.w3.org/Archives/Member/w3c-ac-members/2021JanMar/0054.html
